hydra-node/src/Hydra/Ledger/Cardano/Configuration.hs

```diff
@@ -0,0 +1,83 @@
+module Hydra.Ledger.Cardano.Configuration (
+  module Hydra.Ledger.Cardano.Configuration,
+  Ledger.Globals,
+  Ledger.LedgerEnv,
+) where
+
+import Hydra.Cardano.Api
+import Hydra.Prelude
+
+import qualified Cardano.Ledger.BaseTypes as Ledger
+import qualified Cardano.Ledger.Shelley.Genesis as Ledger
+import qualified Cardano.Ledger.Shelley.LedgerState as Ledger
+import qualified Cardano.Ledger.Shelley.Rules.Ledger as Ledger
+import Cardano.Slotting.EpochInfo (hoistEpochInfo)
+import Cardano.Slotting.Time (mkSlotLength)
+import Control.Arrow (left)
+import Control.Monad.Trans.Except (runExcept)
+import qualified Data.Aeson as Json
+import qualified Data.Aeson.Types as Json
+import qualified Ouroboros.Consensus.HardFork.History as Consensus
+
+-- * Helpers
+
+readJsonFileThrow :: (Json.Value -> Json.Parser a) -> FilePath -> IO a
+readJsonFileThrow parser filepath = do
+  value <- Json.eitherDecodeFileStrict filepath >>= either fail pure
+  case Json.parseEither parser value of
+    Left e -> fail e
+    Right a -> pure a
```

Comment:
Because the latter requires a `ToJSON` instance and does not allow passing an explicit decoder.

Turns out in the end that there exists ToJSON instances for both actually, which I wasn't expecting so it's arguably redundant now.